calculator/main.py

```diff
@@ -37,14 +53,18 @@ def add_new_operation(calc, operation):
     :param operation: Dict with the single operation to be added.
                       ie: {'add': add_function}
     """
-    pass
+    if not isinstance(operation, dict):
+        raise InvalidOperation("Given operation is invalid.")
```

Comment:
This code looks familiar (from `create_new_calculator`)... Can you maybe avoid repetition?